---
title: "Introduction"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Look at your mocks... 

Every microbiome sequencing experiment must have a positive control. However, how do we make use of these mock controls to guide our quality check is not easily available. A basic question to asks, _Is the composition in experimental mock standards similar to theoretical expected composition?_    
We can visually compare the composition bar-plots and check for correlation between experimental and theoretical community composition.  
`chkMocks` eases these basic comparisons.  
Note: Currently, only `ZymoBiomics` mocks are supported and we are working on adding support for custom databases. 

**Install** 
```{r eval=FALSE}

library(devtools) # make sure you have installed devtools
install_github("microsud/chkMocks")

```


```{r setup}
library(chkMocks)
library(dplyr)
library(phyloseq)
library(patchwork)
library(ggplot2)
```

## ZymoBiomics     
Before starting the analysis you need:  

* Raw data processed with dada2 pipeline to get a phyloseq object with `taxa_names` as ASV seqs.     
* If the phyloseq object has samples and mocks, then subset to keep only the mocks.  

Here, the example data are from Karstens L, Asquith M, Davin S, Fair D, Gregory WT, Wolfe AJ, Braun J, McWeeney S. 2019. [Controlling for contaminants in low-biomass 16S rRNA gene sequencing experiments](https://doi.org/10.1128/mSystems.00290-19). mSystems 4:e00290-19.  

```{r}

ZymoExamplePseq

```

```{r}
# check information
sample_data(ZymoExamplePseq)
```


A single function, `checkZymoBiomics` will do the following:  

Take an input phyloseq object of mock communities with `taxa_names` as ASV seqs and use the `ZymoTrainingSet` to assign taxonomy. The `ZymoTrainingSet` contains only the full-length 16S rRNA gene sequences of the candidates in ZymoBIOMICSâ„¢ Microbial Community Standard. [Catalog No. D6300](https://files.zymoresearch.com/protocols/_d6300_zymobiomics_microbial_community_standard.pdf). 
ASVs that are unrelated to `ZymoTrainingSet` are labelled unclassified. The function will return two phyloseq objects a) ASVs level and b) Agglomerated at Species level. The agglomerated species level data is used to check for correlation with theoretical composition.    

```{r}
output.dat <- checkZymoBiomics(ZymoExamplePseq,
                               mock_db = ZymoTrainingSet,
                               multithread= 2,
                               threshold = 60,
                               verbose = FALSE)

cortable <- output.dat$corrTable
colnames(cortable) <- c("MockSampleID", "Correlation2ZymoTheoretical", "MockSampleID_2" )

cortable
```


Get the agglomerated species level data.
```{r}
ps_species <- output.dat$ps_species

ps_species
```

Check assignments
```{r}

get_taxa_unique(output.dat$ps_species, "Species")

```

Plot composition  
```{r fig.height=4, fig.width=8}

p <- plotZymoDefault(output.dat)
p

```

The above plot demonstrates how well/bad the experimental mocks behaved compared to theoretical composition.  
The data from Karstens L, et al., 2019. [mSystems](https://doi.org/10.1128/mSystems.00290-19) started from cells->DNA extraction->pcr amplification->sequencing. At every step there is a possibility for bias. This is demonstrated by differences between the undiluted mock sample and Zymo theoretical composition. Additionally, diluted samples have several unknown, potential contaminants common to low-biomass samples.    

Check how individual taxa were measured.  
```{r fig.width=6, fig.height=5}

p <- plotZymoDefault(output.dat)

# using patchwork plot to extract first bar plot
p[[1]] + facet_wrap(~FeatureID) + 
  theme_minimal(base_size = 10) + 
  theme(legend.position = "none",
        strip.text = element_text(face="italic"))
```

Looking at individual strain abundances indicates under severe counting of _Staphylococcus aureus_. It is also important to note the limitations of species level assignments for short-read length ASVs.  

In the diluted mock samples, there are `Unknown` taxa i.e. those that are not of ZymoBiomics origin.  
Check for their contribution.  
```{r fig.width=6, fig.height=4}
sp.df <- phyloseq::psmelt(ps_species) %>%
  dplyr::filter(species=="Unknown" & Sample !="ZymoTheoretical")

ggplot(sp.df, aes(Sample, Abundance)) + 
  geom_col() + theme_minimal() +
  ggplot2::ylab("Non-Zymo Abundance (%)")+
      ggplot2::xlab("Samples")

```
The lowest dilution, D8 has 80% non-ZymoBiomics taxa, so make sure you have negative controls and check the article by Karstens L, et al. [Controlling for contaminants in low-biomass 16S rRNA gene sequencing experiments](https://doi.org/10.1128/mSystems.00290-19). mSystems 4:e00290-19. All codes available from their analysis are available openly. 


```{r}
devtools::session_info()
```

Disclaimer: While we use `ZymoBiomics` data, we the developers of `chkMocks` are not associated with the manufacturers and this work should not be considered as an endorsement for the said product.    
