<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction • chkMocks</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="chkMocks">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">chkMocks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.01</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/aIntroduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/bCreateCustomDatabase.html">Create a custome database</a>
    </li>
    <li>
      <a href="../articles/cExampleCustomMocks.html">Working with custom mocks</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microsud/chkMocks/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="aIntroduction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/microsud/chkMocks/blob/master/vignettes/aIntroduction.Rmd"><code>vignettes/aIntroduction.Rmd</code></a></small>
      <div class="hidden name"><code>aIntroduction.Rmd</code></div>

    </div>

    
    
<p>Look at your mocks…</p>
<p>Every microbiome sequencing experiment must have a positive control. However, how do we make use of these mock controls to guide our quality check is not easily available. A basic question to asks, <em>Is the composition in experimental mock standards similar to theoretical expected composition?</em><br>
We can visually compare the composition bar-plots and check for correlation between experimental and theoretical community composition.<br><code>chkMocks</code> eases these basic comparisons.<br>
Note: Currently, only <code>ZymoBiomics</code> mocks are supported and we are working on adding support for custom databases.</p>
<p><strong>Install</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span> <span class="co"># make sure you have installed devtools</span>
<span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"microsud/chkMocks"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microsud/chkMocks">chkMocks</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'patchwork' was built under R version 4.0.5</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span></code></pre></div>
<div id="zymobiomics" class="section level2">
<h2 class="hasAnchor">
<a href="#zymobiomics" class="anchor"></a>ZymoBiomics</h2>
<p>Before starting the analysis you need:</p>
<ul>
<li>Raw data processed with dada2 pipeline to get a phyloseq object with <code>taxa_names</code> as ASV seqs.<br>
</li>
<li>If the phyloseq object has samples and mocks, then subset to keep only the mocks.</li>
</ul>
<p>Here, the example data are from Karstens L, Asquith M, Davin S, Fair D, Gregory WT, Wolfe AJ, Braun J, McWeeney S. 2019. <a href="https://doi.org/10.1128/mSystems.00290-19">Controlling for contaminants in low-biomass 16S rRNA gene sequencing experiments</a>. mSystems 4:e00290-19.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ZymoExamplePseq</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 946 taxa and 9 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 9 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 946 taxa by 6 taxonomic ranks ]</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># check information</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ZymoExamplePseq</span><span class="op">)</span>
<span class="co">#&gt;    X.SampleID SampleDescription DNA_conc    SampleType Description is.neg</span>
<span class="co">#&gt; D0         D0           MC_Neat    138.8 MockCommunity        Mock  FALSE</span>
<span class="co">#&gt; D1         D1            MC_1:3    109.2 MockCommunity        Mock  FALSE</span>
<span class="co">#&gt; D2         D2            MC_1:9    105.8 MockCommunity        Mock  FALSE</span>
<span class="co">#&gt; D3         D3           MC_1:27     60.6 MockCommunity        Mock  FALSE</span>
<span class="co">#&gt; D4         D4           MC_1:81     34.9 MockCommunity        Mock  FALSE</span>
<span class="co">#&gt; D5         D5          MC_1:243     22.6 MockCommunity        Mock  FALSE</span>
<span class="co">#&gt; D6         D6          MC_1:729     20.0 MockCommunity        Mock  FALSE</span>
<span class="co">#&gt; D7         D7         MC_1:2187     11.4 MockCommunity        Mock  FALSE</span>
<span class="co">#&gt; D8         D8         MC_1:6561     13.8 MockCommunity        Mock  FALSE</span></code></pre></div>
<p>A single function, <code>checkZymoBiomics</code> will do the following:</p>
<p>Take an input phyloseq object of mock communities with <code>taxa_names</code> as ASV seqs and use the <code>ZymoTrainingSet</code> to assign taxonomy. The <code>ZymoTrainingSet</code> contains only the full-length 16S rRNA gene sequences of the candidates in ZymoBIOMICS™ Microbial Community Standard. <a href="https://files.zymoresearch.com/protocols/_d6300_zymobiomics_microbial_community_standard.pdf">Catalog No. D6300</a>. ASVs that are unrelated to <code>ZymoTrainingSet</code> are labelled unclassified. The function will return two phyloseq objects a) ASVs level and b) Agglomerated at Species level. The agglomerated species level data is used to check for correlation with theoretical composition.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">output.dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkZymoBiomics.html">checkZymoBiomics</a></span><span class="op">(</span><span class="va">ZymoExamplePseq</span>,
                               mock_db <span class="op">=</span> <span class="va">ZymoTrainingSet</span>,
                               multithread<span class="op">=</span> <span class="fl">2</span>,
                               threshold <span class="op">=</span> <span class="fl">60</span>,
                               verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; Using internal reference database</span>
<span class="co">#&gt;             for ZymoBiomics</span>
<span class="co">#&gt; Warning: Expected 8 pieces. Missing pieces filled with `NA` in 935 rows [9, 11,</span>
<span class="co">#&gt; 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, ...].</span>

<span class="va">cortable</span> <span class="op">&lt;-</span> <span class="va">output.dat</span><span class="op">$</span><span class="va">corrTable</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">cortable</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MockSampleID"</span>, <span class="st">"Correlation2ZymoTheoretical"</span>, <span class="st">"MockSampleID_2"</span> <span class="op">)</span>

<span class="va">cortable</span>
<span class="co">#&gt; # A tibble: 10 x 3</span>
<span class="co">#&gt;    MockSampleID    Correlation2ZymoTheoretical MockSampleID_2 </span>
<span class="co">#&gt;    &lt;chr&gt;                                 &lt;dbl&gt; &lt;fct&gt;          </span>
<span class="co">#&gt;  1 D0                                   0.233  D0             </span>
<span class="co">#&gt;  2 D1                                   0.483  D1             </span>
<span class="co">#&gt;  3 D2                                   0.383  D2             </span>
<span class="co">#&gt;  4 D3                                   0.35   D3             </span>
<span class="co">#&gt;  5 D4                                   0.367  D4             </span>
<span class="co">#&gt;  6 D5                                  -0.117  D5             </span>
<span class="co">#&gt;  7 D6                                   0.05   D6             </span>
<span class="co">#&gt;  8 D7                                   0      D7             </span>
<span class="co">#&gt;  9 D8                                   0.0833 D8             </span>
<span class="co">#&gt; 10 ZymoTheoretical                      1      ZymoTheoretical</span></code></pre></div>
<p>Get the agglomerated species level data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps_species</span> <span class="op">&lt;-</span> <span class="va">output.dat</span><span class="op">$</span><span class="va">ps_species</span>

<span class="va">ps_species</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 9 taxa and 10 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 10 samples by 8 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 9 taxa by 15 taxonomic ranks ]</span></code></pre></div>
<p>Check assignments</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">output.dat</span><span class="op">$</span><span class="va">ps_species</span>, <span class="st">"Species"</span><span class="op">)</span>
<span class="co">#&gt; [1] "Bacillus.subtilis"       "Listeria.monocytogenes" </span>
<span class="co">#&gt; [3] "Staphylococcus.aureus"   "Enterococcus.faecalis"  </span>
<span class="co">#&gt; [5] "Lactobacillus.fermentum" "Escherichia.coli"       </span>
<span class="co">#&gt; [7] "Salmonella.enterica"     "Pseudomonas.aeruginosa" </span>
<span class="co">#&gt; [9] NA</span></code></pre></div>
<p>Plot composition</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotZymoDefault.html">plotZymoDefault</a></span><span class="op">(</span><span class="va">output.dat</span><span class="op">)</span>
<span class="va">p</span></code></pre></div>
<p><img src="aIntroduction_files/figure-html/unnamed-chunk-8-1.png" width="2400"></p>
<p>The above plot demonstrates how well/bad the experimental mocks behaved compared to theoretical composition.<br>
The data from Karstens L, et al., 2019. <a href="https://doi.org/10.1128/mSystems.00290-19">mSystems</a> started from cells-&gt;DNA extraction-&gt;pcr amplification-&gt;sequencing. At every step there is a possibility for bias. This is demonstrated by differences between the undiluted mock sample and Zymo theoretical composition. Additionally, diluted samples have several unknown, potential contaminants common to low-biomass samples.</p>
<p>Check how individual taxa were measured.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotZymoDefault.html">plotZymoDefault</a></span><span class="op">(</span><span class="va">output.dat</span><span class="op">)</span>

<span class="co"># using patchwork plot to extract first bar plot</span>
<span class="va">p</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">FeatureID</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>,
        strip.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face<span class="op">=</span><span class="st">"italic"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="aIntroduction_files/figure-html/unnamed-chunk-9-1.png" width="1800"></p>
<p>Looking at individual strain abundances indicates under counting of <em>Staphylococcus aureus</em>. It is also important to note the limitations of species level assignments for short-read length ASVs.</p>
<p>In the diluted mock samples, there are <code>Unknown</code> taxa i.e. those that are not of ZymoBiomics origin.<br>
Check for their contribution.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sp.df</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/psmelt.html">psmelt</a></span><span class="op">(</span><span class="va">ps_species</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">species</span><span class="op">==</span><span class="st">"Unknown"</span> <span class="op">&amp;</span> <span class="va">Sample</span> <span class="op">!=</span><span class="st">"ZymoTheoretical"</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">sp.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">Sample</span>, <span class="va">Abundance</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Non-Zymo Abundance (%)"</span><span class="op">)</span><span class="op">+</span>
      <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Samples"</span><span class="op">)</span></code></pre></div>
<p><img src="aIntroduction_files/figure-html/unnamed-chunk-10-1.png" width="1800"> The lowest dilution, D8 has 80% non-ZymoBiomics taxa, so make sure you have negative controls and check the article by Karstens L, et al. <a href="https://doi.org/10.1128/mSystems.00290-19">Controlling for contaminants in low-biomass 16S rRNA gene sequencing experiments</a>. mSystems 4:e00290-19. All codes from their analysis are available openly. Note: The taxonomy of lactobacilli has been updated from <em>L. fermentum</em> is <em>Limosilactobacillus</em> <a href="https://doi.org/10.1099/ijsem.0.004107">Zheng J., Wittouck S., Salvetti E. et al.,(2020). A taxonomic note on the genus <em>Lactobacillus</em>: Description of 23 novel genera, emended description of the genus <em>Lactobacillus</em> Beijerink 1901, and union of <em>Lactobacillaceae</em> and <em>Leuconostocaceae</em>.</a><br>
Thanks to <a href="https://twitter.com/FelisGiovanna">Giovanna Felis</a> for bringing this to my notice on twitter.<br>
The label for <em>L. fermentum</em> is still being used by ZymoBiomics and will keep it as it is for now.</p>
<p><strong>Other independently developed tools that are:</strong><br>
ZymoResearch <a href="https://github.com/Zymo-Research/miqScore16SPublic">miqScore16SPublic</a> by <a href="https://twitter.com/bionomicon">Michael Weinstein</a><br>
QIIME2 <a href="https://github.com/qiime2/q2-quality-control">q2-quality-control</a> suggested by <a href="https://twitter.com/li_yanxian">Yanxian Li</a><br>
OCMS <a href="https://github.com/OxfordCMS/OCMS_zymoBIOMICS">OCMS_zymoBIOMICS</a> by <a href="https://twitter.com/IlottNick">Nick Ilott</a></p>
<p>Let me know if there are more tools that need to be mentioned here <a href="https://github.com/microsud/chkMocks/issues">GitHub issues</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; - Session info ---------------------------------------------------------------</span>
<span class="co">#&gt;  setting  value                       </span>
<span class="co">#&gt;  version  R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt;  os       Windows 10 x64              </span>
<span class="co">#&gt;  system   x86_64, mingw32             </span>
<span class="co">#&gt;  ui       RTerm                       </span>
<span class="co">#&gt;  language (EN)                        </span>
<span class="co">#&gt;  collate  English_Netherlands.1252    </span>
<span class="co">#&gt;  ctype    English_Netherlands.1252    </span>
<span class="co">#&gt;  tz       Europe/Berlin               </span>
<span class="co">#&gt;  date     2021-05-22                  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; - Packages -------------------------------------------------------------------</span>
<span class="co">#&gt;  ! package              * version  date       lib source        </span>
<span class="co">#&gt;    ade4                   1.7-16   2020-10-28 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    ape                    5.4-1    2020-08-13 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    assertthat             0.2.1    2019-03-21 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    Biobase                2.50.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    BiocGenerics           0.36.1   2021-04-16 [3] Bioconductor  </span>
<span class="co">#&gt;    BiocParallel           1.24.1   2020-11-06 [3] Bioconductor  </span>
<span class="co">#&gt;    biomformat             1.18.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    Biostrings             2.58.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    bit                    4.0.4    2020-08-04 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    bit64                  4.0.5    2020-08-30 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    bitops                 1.0-6    2013-08-17 [3] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    blob                   1.2.1    2020-01-20 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    bslib                  0.2.4    2021-01-25 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    cachem                 1.0.4    2021-02-13 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    callr                  3.7.0    2021-04-20 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    chkMocks             * 0.1.01   2021-05-22 [1] local         </span>
<span class="co">#&gt;    cli                    2.4.0    2021-04-05 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    cluster                2.1.0    2019-06-19 [4] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    codetools              0.2-16   2018-12-24 [4] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    colorspace             1.4-1    2019-03-18 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    corrr                  0.4.3    2020-11-24 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    crayon                 1.4.1    2021-02-08 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    dada2                  1.18.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    data.table             1.13.0   2020-07-24 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    DBI                    1.1.1    2021-01-15 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    DECIPHER               2.18.1   2020-10-29 [3] Bioconductor  </span>
<span class="co">#&gt;    DelayedArray           0.16.3   2021-03-24 [3] Bioconductor  </span>
<span class="co">#&gt;    desc                   1.3.0    2021-03-05 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    devtools               2.4.0    2021-04-07 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    digest                 0.6.27   2020-10-24 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    dplyr                * 1.0.5    2021-03-05 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    ellipsis               0.3.1    2020-05-15 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    evaluate               0.14     2019-05-28 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    fansi                  0.4.2    2021-01-15 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    farver                 2.1.0    2021-02-28 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    fastmap                1.0.1    2019-10-08 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    foreach                1.5.1    2020-10-15 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    fs                     1.5.0    2020-07-31 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    generics               0.1.0    2020-10-31 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    GenomeInfoDb           1.26.7   2021-04-09 [3] Bioconductor  </span>
<span class="co">#&gt;    GenomeInfoDbData       1.2.4    2020-10-19 [3] Bioconductor  </span>
<span class="co">#&gt;    GenomicAlignments      1.26.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    GenomicRanges          1.42.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    ggplot2              * 3.3.3    2020-12-30 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    glue                   1.4.2    2020-08-27 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    gtable                 0.3.0    2019-03-25 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    highr                  0.9      2021-04-16 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    hms                    1.0.0    2021-01-13 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    htmltools              0.5.1.1  2021-01-22 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    hwriter                1.3.2    2014-09-10 [3] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    igraph                 1.2.6    2020-10-06 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    IRanges                2.24.1   2020-12-12 [3] Bioconductor  </span>
<span class="co">#&gt;    iterators              1.0.13   2020-10-15 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    jpeg                   0.1-8.1  2019-10-24 [3] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    jquerylib              0.1.3    2020-12-17 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    jsonlite               1.7.1    2020-09-07 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    knitr                  1.32     2021-04-14 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    labeling               0.4.2    2020-10-20 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    lattice                0.20-41  2020-04-02 [4] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    latticeExtra           0.6-29   2019-12-19 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    lifecycle              1.0.0    2021-02-15 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    magrittr               2.0.1    2020-11-17 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    MASS                   7.3-53   2020-09-09 [4] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    Matrix                 1.2-18   2019-11-27 [4] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    MatrixGenerics         1.2.1    2021-01-30 [3] Bioconductor  </span>
<span class="co">#&gt;    matrixStats            0.58.0   2021-01-29 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    memoise                2.0.0    2021-01-26 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    mgcv                   1.8-33   2020-08-27 [4] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    microbiome             1.12.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    multtest               2.46.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    munsell                0.5.0    2018-06-12 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    nlme                   3.1-149  2020-08-23 [4] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    patchwork            * 1.1.1    2020-12-17 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    permute                0.9-5    2019-03-12 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    phyloseq             * 1.34.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    pillar                 1.6.0    2021-04-13 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    pkgbuild               1.2.0    2020-12-15 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    pkgconfig              2.0.3    2019-09-22 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    pkgdown                1.6.1    2020-09-12 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    pkgload                1.2.1    2021-04-06 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    plyr                   1.8.6    2020-03-03 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    png                    0.1-7    2013-12-03 [3] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    prettyunits            1.1.1    2020-01-24 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    processx               3.5.1    2021-04-04 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    progress               1.2.2    2019-05-16 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    ps                     1.6.0    2021-02-28 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    purrr                  0.3.4    2020-04-17 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    R6                     2.5.0    2020-10-28 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    ragg                   1.1.2    2021-03-17 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    RColorBrewer           1.1-2    2014-12-07 [3] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    Rcpp                   1.0.5    2020-07-06 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;  D RcppParallel           5.1.2    2021-04-15 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    RCurl                  1.98-1.3 2021-03-16 [3] CRAN (R 4.0.4)</span>
<span class="co">#&gt;    remotes                2.3.0    2021-04-01 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    reshape2               1.4.4    2020-04-09 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    rhdf5                  2.34.0   2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;  D rhdf5filters           1.2.0    2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    Rhdf5lib               1.12.1   2021-01-26 [3] Bioconductor  </span>
<span class="co">#&gt;    rlang                  0.4.10   2020-12-30 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    rmarkdown              2.7      2021-02-19 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    rprojroot              2.0.2    2020-11-15 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    Rsamtools              2.6.0    2020-10-27 [3] Bioconductor  </span>
<span class="co">#&gt;    RSQLite                2.2.1    2020-09-30 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    rstudioapi             0.13     2020-11-12 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    Rtsne                  0.15     2018-11-10 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    S4Vectors              0.28.1   2020-12-09 [3] Bioconductor  </span>
<span class="co">#&gt;    sass                   0.3.1    2021-01-24 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    scales                 1.1.1    2020-05-11 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    sessioninfo            1.1.1    2018-11-05 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    ShortRead              1.48.0   2020-10-28 [3] Bioconductor  </span>
<span class="co">#&gt;    stringi                1.5.3    2020-09-09 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    stringr                1.4.0    2019-02-10 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    SummarizedExperiment   1.20.0   2020-10-28 [3] Bioconductor  </span>
<span class="co">#&gt;    survival               3.2-7    2020-09-28 [4] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    systemfonts            1.0.1    2021-02-09 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    testthat               3.0.2    2021-02-14 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    textshaping            0.3.3    2021-03-16 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    tibble                 3.0.4    2020-10-12 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    tidyr                  1.1.2    2020-08-27 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    tidyselect             1.1.0    2020-05-11 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    usethis                2.0.1    2021-02-10 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    utf8                   1.2.1    2021-03-12 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    vctrs                  0.3.8    2021-04-29 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    vegan                  2.5-7    2020-11-28 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    viridisLite            0.4.0    2021-04-13 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    withr                  2.4.2    2021-04-18 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    xfun                   0.22     2021-03-11 [3] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    XVector                0.30.0   2020-10-28 [3] Bioconductor  </span>
<span class="co">#&gt;    yaml                   2.2.1    2020-02-01 [3] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    zlibbioc               1.36.0   2020-10-28 [3] Bioconductor  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [1] C:/Users/sudar/AppData/Local/Temp/RtmpuwxWnr/temp_libpath163466f4872</span>
<span class="co">#&gt; [2] C:/Users/sudar/AppData/Local/Temp/RtmpuwxWnr/temp_libpath163442811331</span>
<span class="co">#&gt; [3] C:/Users/sudar/Documents/R/win-library/4.0</span>
<span class="co">#&gt; [4] C:/Program Files/R/R-4.0.3/library</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  D -- DLL MD5 mismatch, broken installation.</span></code></pre></div>
<p>Disclaimer: While we use <code>ZymoBiomics</code> data, we the developers of <code>chkMocks</code> are not associated with the manufacturers and this work should not be considered as an endorsement for the said product.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sudarshan Shetty.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
