<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Working with custom mocks • chkMocks</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Working with custom mocks">
<meta property="og:description" content="chkMocks">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">chkMocks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.01</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/aIntroduction.html">Introduction</a>
    </li>
    <li>
      <a href="../articles/bCreateCustomDatabase.html">Create a custome database</a>
    </li>
    <li>
      <a href="../articles/cExampleCustomMocks.html">Working with custom mocks</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microsud/chkMocks/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cExampleCustomMocks_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Working with custom mocks</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/microsud/chkMocks/blob/master/vignettes/cExampleCustomMocks.Rmd"><code>vignettes/cExampleCustomMocks.Rmd</code></a></small>
      <div class="hidden name"><code>cExampleCustomMocks.Rmd</code></div>

    </div>

    
    
<p>In this examples, the step to work with custom mock communities is demonstrated.<br>
The test data used in this example are from Ramiro-Garcia J, Hermes GDA, Giatsis C et al. <a href="https://doi.org/10.12688/f1000research.9227.2">NG-Tax, a highly accurate and validated pipeline for analysis of 16S rRNA amplicons from complex biomes F1000Research 2018, 5:1791</a> [version 2; peer review: 2 approved, 1 approved with reservations, 1 not approved].</p>
<p>How to prepare a custom training set for <code><a href="https://rdrr.io/pkg/DECIPHER/man/IdTaxa.html">DECIPHER::IdTaxa</a></code>?<br>
These are from the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/DECIPHER/inst/doc/ClassifySequences.pdf">DECIPHER</a> package.</p>
<div id="library" class="section level1">
<h1 class="hasAnchor">
<a href="#library" class="anchor"></a>Library</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Following packages are required  </span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microsud/chkMocks">chkMocks</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">DECIPHER</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://microbiome.github.io/microbiome">microbiome</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tidymodels/corrr">corrr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings">Biostrings</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape">reshape2</a></span><span class="op">)</span></code></pre></div>
<p>Read the fasta file with 16S rRNA gene sequences for taxa in mock community.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># db &lt;- "../MIBMocks/MIBPhylotypes.fasta" # path of custom fasta sequences </span>

<span class="co"># Here example is store in the package.</span>
<span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"MIBPhylotypes.fasta"</span>,
                      package<span class="op">=</span><span class="st">"chkMocks"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Convert to <code>DNAStringSet</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">seqs</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readDNAStringSet</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span>
<span class="co">#&gt; Warning in .Call2("fasta_index", filexp_list, nrec, skip, seek.first.rec, :</span>
<span class="co">#&gt; reading FASTA file C:/Users/sudar/AppData/Local/Temp/RtmpuwxWnr/</span>
<span class="co">#&gt; temp_libpath163466f4872/chkMocks/extdata/MIBPhylotypes.fasta: ignored 2 invalid</span>
<span class="co">#&gt; one-letter sequence codes</span>
<span class="va">seqs</span> <span class="op">&lt;-</span> <span class="fu">DECIPHER</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/OrientNucleotides.html">OrientNucleotides</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span>
<span class="co">#&gt; ========================================================================================================================================================================================================</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Time difference of 0.24 secs</span>
<span class="co"># check first 2 as example</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>
<span class="co">#&gt; [1] "Bacteria;Actinobacteria;Actinobacteria;Actinomycetales;Nocardiaceae;Rhodococcus;Rhodococcus.1_PT;"  </span>
<span class="co">#&gt; [2] "Bacteria;Actinobacteria;Actinobacteria;Actinomycetales;Micrococcaceae;Micrococcus;Micrococcus.2_PT;"</span></code></pre></div>
<p>Now, adding the dummy seq name before &gt; and adding a ‘Root’ before Bacteria;Phylum;etc;</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># here, adding the dummy seq name before &gt; and adding a 'Root' before Bacteria;Phylum;etc;</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"MIB"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="st">" "</span>, <span class="st">"Root;"</span> ,<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span><span class="op">)</span>

<span class="co"># check first 2 as example</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>
<span class="co">#&gt; [1] "MIB1 Root;Bacteria;Actinobacteria;Actinobacteria;Actinomycetales;Nocardiaceae;Rhodococcus;Rhodococcus.1_PT;"  </span>
<span class="co">#&gt; [2] "MIB2 Root;Bacteria;Actinobacteria;Actinobacteria;Actinomycetales;Micrococcaceae;Micrococcus;Micrococcus.2_PT;"</span></code></pre></div>
<p>We show how to check for problematic taxonomies. However, it will most likely be not required if input fasta is formatted correctly.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span> <span class="co"># sequence names</span>
<span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"(.*)(Root;)"</span>, <span class="st">"\\2"</span>, <span class="va">groups</span><span class="op">)</span>
<span class="va">groupCounts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">groups</span><span class="op">)</span>
<span class="va">u_groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">groupCounts</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">u_groups</span><span class="op">)</span>
<span class="va">maxGroupSize</span> <span class="op">&lt;-</span> <span class="fl">10</span> <span class="co"># max sequences per label (&gt;= 1)</span>
<span class="va">remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/logical.html">logical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>Create a training set.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">maxIterations</span> <span class="op">&lt;-</span> <span class="fl">3</span>
<span class="va">allowGroupRemoval</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span>
<span class="va">probSeqsPrev</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html">integer</a></span><span class="op">(</span><span class="op">)</span>

<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">groupCounts</span> <span class="op">&gt;</span> <span class="va">maxGroupSize</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="va">index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">groups</span><span class="op">==</span><span class="va">u_groups</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>
  <span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">index</span><span class="op">)</span>,
                 <span class="va">maxGroupSize</span><span class="op">)</span>
  <span class="va">remove</span><span class="op">[</span><span class="va">index</span><span class="op">[</span><span class="op">-</span><span class="va">keep</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span>
<span class="op">}</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">remove</span><span class="op">)</span>

<span class="va">taxid</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="va">maxIterations</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Training iteration: "</span>, <span class="va">i</span>, <span class="st">"\n"</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span>
  <span class="co"># train the classifier</span>
  <span class="va">MIBTrainingSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/LearnTaxa.html">LearnTaxa</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">[</span><span class="op">!</span><span class="va">remove</span><span class="op">]</span>,
                              <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">seqs</span><span class="op">)</span><span class="op">[</span><span class="op">!</span><span class="va">remove</span><span class="op">]</span>,
                              <span class="va">taxid</span><span class="op">)</span>
  <span class="co"># look for problem sequences</span>
  <span class="va">probSeqs</span> <span class="op">&lt;-</span> <span class="va">MIBTrainingSet</span><span class="op">$</span><span class="va">problemSequences</span><span class="op">$</span><span class="va">Index</span> 
  <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">probSeqs</span><span class="op">)</span><span class="op">==</span><span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"No problem sequences remaining.\n"</span><span class="op">)</span>
    <span class="kw">break</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">probSeqs</span><span class="op">)</span><span class="op">==</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">probSeqsPrev</span><span class="op">)</span> <span class="op">&amp;&amp;</span>
             <span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="va">probSeqsPrev</span><span class="op">==</span><span class="va">probSeqs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="st">"Iterations converged.\n"</span><span class="op">)</span>
    <span class="kw">break</span>
  <span class="op">}</span>
  <span class="kw">if</span> <span class="op">(</span><span class="va">i</span><span class="op">==</span><span class="va">maxIterations</span><span class="op">)</span>
    <span class="kw">break</span>
  <span class="va">probSeqsPrev</span> <span class="op">&lt;-</span> <span class="va">probSeqs</span>
  <span class="co"># remove any problem sequences</span>
  <span class="va">index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="op">!</span><span class="va">remove</span><span class="op">)</span><span class="op">[</span><span class="va">probSeqs</span><span class="op">]</span>
  <span class="va">remove</span><span class="op">[</span><span class="va">index</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span> <span class="co"># remove all problem sequences</span>
  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="va">allowGroupRemoval</span><span class="op">)</span> <span class="op">{</span>
    <span class="co"># replace any removed groups</span>
    <span class="va">missing</span> <span class="op">&lt;-</span> <span class="op">!</span><span class="op">(</span><span class="va">u_groups</span> <span class="op">%in%</span> <span class="va">groups</span><span class="op">[</span><span class="op">!</span><span class="va">remove</span><span class="op">]</span><span class="op">)</span>
    <span class="va">missing</span> <span class="op">&lt;-</span> <span class="va">u_groups</span><span class="op">[</span><span class="va">missing</span><span class="op">]</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">missing</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span>
      <span class="va">index</span> <span class="op">&lt;-</span> <span class="va">index</span><span class="op">[</span><span class="va">groups</span><span class="op">[</span><span class="va">index</span><span class="op">]</span> <span class="op">%in%</span> <span class="va">missing</span><span class="op">]</span>
      <span class="va">remove</span><span class="op">[</span><span class="va">index</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span> <span class="co"># don't remove</span>
    <span class="op">}</span>
  <span class="op">}</span>
<span class="op">}</span></code></pre></div>
<p>Check any problems.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">remove</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">probSeqs</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># saveRDS(MIBTrainingSet, "inst/extdata/MIBTrainingSet.rds")</span>
<span class="co"># Here MIBTrainingSet is stored in the package to reduce time for example.</span>
<span class="co"># Read data from package</span>
<span class="va">MIBTrainingSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"MIBTrainingSet.rds"</span>,
                      package<span class="op">=</span><span class="st">"chkMocks"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#path for file</span>
<span class="va">MIBTrainingSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">MIBTrainingSet</span><span class="op">)</span></code></pre></div>
<p>Check training set.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">MIBTrainingSet</span><span class="op">)</span></code></pre></div>
<p><img src="cExampleCustomMocks_files/figure-html/plot-training-1.png" width="3000"></p>
<p>Check training set.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Problem seqs in reference?</span>
<span class="va">MIBTrainingSet</span><span class="op">$</span><span class="va">problemSequences</span>
<span class="co">#&gt;   Index</span>
<span class="co">#&gt; 1    16</span>
<span class="co">#&gt;                                                                                              Expected</span>
<span class="co">#&gt; 1 Root;Bacteria;Bacteroidetes;Bacteroidia;Bacteroidales;Bacteroidaceae;Bacteroides;Bacteroides.16_PT;</span>
<span class="co">#&gt;                                                                                             Predicted</span>
<span class="co">#&gt; 1 Root;Bacteria;Bacteroidetes;Bacteroidia;Bacteroidales;Bacteroidaceae;Bacteroides;Bacteroides.11_PT;</span>

<span class="co"># Which is the problem group</span>
<span class="va">MIBTrainingSet</span><span class="op">$</span><span class="va">problemGroups</span>
<span class="co">#&gt; [1] "Root;Bacteria;Bacteroidetes;Bacteroidia;Bacteroidales;Bacteroidaceae;Bacteroides;"</span></code></pre></div>
<p><code>Bacteroides</code> seqs are highlighted here. These are common gut inhabitants. Closely related <code>Bacteroides</code> can be difficult to assign taxonomy at lowere levels.</p>
<p>There is <code>N</code> in some of the reference sequences and therefore it is highlighted.</p>
<div id="create-theoretical-phyloseq" class="section level2">
<h2 class="hasAnchor">
<a href="#create-theoretical-phyloseq" class="anchor"></a>Create theoretical Phyloseq</h2>
<div id="get-theoretical-composition-mibmocks" class="section level3">
<h3 class="hasAnchor">
<a href="#get-theoretical-composition-mibmocks" class="anchor"></a>Get theoretical Composition MIBMocks</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># mck.otu.th.path &lt;- read.csv("../MIBMocks/TheoreticalCompositionMIBMocks.csv") </span>

<span class="co"># Here example is store in the package.</span>
<span class="va">mck.otu.th.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TheoreticalCompositionMIBMocks.csv"</span>,
                      package<span class="op">=</span><span class="st">"chkMocks"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">mck.otu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">mck.otu.th.path</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mck.otu</span><span class="op">)</span>
<span class="co">#&gt;                Species  MC3   MC4</span>
<span class="co">#&gt; 1     Rhodococcus.1_PT 1.16 0.001</span>
<span class="co">#&gt; 2     Micrococcus.2_PT 1.16 0.010</span>
<span class="co">#&gt; 3 Bifidobacterium.3_PT 1.16 0.100</span>
<span class="co">#&gt; 4 Bifidobacterium.4_PT 1.16 2.490</span>
<span class="co">#&gt; 5 Bifidobacterium.5_PT 1.16 0.000</span>
<span class="co">#&gt; 6 Bifidobacterium.6_PT 1.16 0.000</span></code></pre></div>
<p>The table above has theoretical composition.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># make Species col as rownames</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">mck.otu</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">mck.otu</span><span class="op">$</span><span class="va">Species</span>

<span class="co"># Remove first col `Species` and convert it to a matrix</span>
<span class="va">mck.otu</span> <span class="op">&lt;-</span> <span class="va">mck.otu</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="op">)</span> 

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mck.otu</span><span class="op">)</span>
<span class="co">#&gt;                       MC3   MC4</span>
<span class="co">#&gt; Rhodococcus.1_PT     1.16 0.001</span>
<span class="co">#&gt; Micrococcus.2_PT     1.16 0.010</span>
<span class="co">#&gt; Bifidobacterium.3_PT 1.16 0.100</span>
<span class="co">#&gt; Bifidobacterium.4_PT 1.16 2.490</span>
<span class="co">#&gt; Bifidobacterium.5_PT 1.16 0.000</span>
<span class="co">#&gt; Bifidobacterium.6_PT 1.16 0.000</span></code></pre></div>
<p>The matrix above can be convert to otu_table later.</p>
<p>Now create a dummy <code>sample_ _data</code> table.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># SampleType here is label that should match one of your columns in sample_data in experimental samples phyloseq object.</span>
<span class="va">mck.sam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">mck.otu</span><span class="op">)</span><span class="op">)</span>,
                      SampleType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"MyMockTheoretical"</span>,<span class="st">"MyMockTheoretical"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>  
  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">mck.sam</span>
<span class="co">#&gt;            SampleType</span>
<span class="co">#&gt; MC3 MyMockTheoretical</span>
<span class="co">#&gt; MC4 MyMockTheoretical</span></code></pre></div>
<p>Get the taxonomy for mock phylotype.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># mck.taxonomy.th.path &lt;- read.csv("../MIBMocks/TaxonomyMIBMocks.csv") </span>

<span class="co"># Here example is store in the package.</span>
<span class="va">mck.taxonomy.th.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TaxonomyMIBMocks.csv"</span>,
                      package<span class="op">=</span><span class="st">"chkMocks"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">mck.tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span><span class="op">(</span><span class="va">mck.taxonomy.th.path</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mck.tax</span><span class="op">)</span>
<span class="co">#&gt;     Domain         Phylum          Class             Order             Family</span>
<span class="co">#&gt; 1 Bacteria Actinobacteria Actinobacteria   Actinomycetales       Nocardiaceae</span>
<span class="co">#&gt; 2 Bacteria Actinobacteria Actinobacteria   Actinomycetales     Micrococcaceae</span>
<span class="co">#&gt; 3 Bacteria Actinobacteria Actinobacteria Bifidobacteriales Bifidobacteriaceae</span>
<span class="co">#&gt; 4 Bacteria Actinobacteria Actinobacteria Bifidobacteriales Bifidobacteriaceae</span>
<span class="co">#&gt; 5 Bacteria Actinobacteria Actinobacteria Bifidobacteriales Bifidobacteriaceae</span>
<span class="co">#&gt; 6 Bacteria Actinobacteria Actinobacteria Bifidobacteriales Bifidobacteriaceae</span>
<span class="co">#&gt;             Genus              Species</span>
<span class="co">#&gt; 1     Rhodococcus     Rhodococcus.1_PT</span>
<span class="co">#&gt; 2     Micrococcus     Micrococcus.2_PT</span>
<span class="co">#&gt; 3 Bifidobacterium Bifidobacterium.3_PT</span>
<span class="co">#&gt; 4 Bifidobacterium Bifidobacterium.4_PT</span>
<span class="co">#&gt; 5 Bifidobacterium Bifidobacterium.5_PT</span>
<span class="co">#&gt; 6 Bifidobacterium Bifidobacterium.6_PT</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">mck.tax</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">mck.tax</span><span class="op">$</span><span class="va">Species</span>
<span class="va">mck.tax</span> <span class="op">&lt;-</span> <span class="va">mck.tax</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span><span class="op">(</span><span class="op">)</span> 

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">mck.tax</span><span class="op">)</span>
<span class="co">#&gt;                      Phylum           Class            Order              </span>
<span class="co">#&gt; Rhodococcus.1_PT     "Actinobacteria" "Actinobacteria" "Actinomycetales"  </span>
<span class="co">#&gt; Micrococcus.2_PT     "Actinobacteria" "Actinobacteria" "Actinomycetales"  </span>
<span class="co">#&gt; Bifidobacterium.3_PT "Actinobacteria" "Actinobacteria" "Bifidobacteriales"</span>
<span class="co">#&gt; Bifidobacterium.4_PT "Actinobacteria" "Actinobacteria" "Bifidobacteriales"</span>
<span class="co">#&gt; Bifidobacterium.5_PT "Actinobacteria" "Actinobacteria" "Bifidobacteriales"</span>
<span class="co">#&gt; Bifidobacterium.6_PT "Actinobacteria" "Actinobacteria" "Bifidobacteriales"</span>
<span class="co">#&gt;                      Family               Genus            </span>
<span class="co">#&gt; Rhodococcus.1_PT     "Nocardiaceae"       "Rhodococcus"    </span>
<span class="co">#&gt; Micrococcus.2_PT     "Micrococcaceae"     "Micrococcus"    </span>
<span class="co">#&gt; Bifidobacterium.3_PT "Bifidobacteriaceae" "Bifidobacterium"</span>
<span class="co">#&gt; Bifidobacterium.4_PT "Bifidobacteriaceae" "Bifidobacterium"</span>
<span class="co">#&gt; Bifidobacterium.5_PT "Bifidobacteriaceae" "Bifidobacterium"</span>
<span class="co">#&gt; Bifidobacterium.6_PT "Bifidobacteriaceae" "Bifidobacterium"</span>
<span class="co">#&gt;                      Species               </span>
<span class="co">#&gt; Rhodococcus.1_PT     "Rhodococcus.1_PT"    </span>
<span class="co">#&gt; Micrococcus.2_PT     "Micrococcus.2_PT"    </span>
<span class="co">#&gt; Bifidobacterium.3_PT "Bifidobacterium.3_PT"</span>
<span class="co">#&gt; Bifidobacterium.4_PT "Bifidobacterium.4_PT"</span>
<span class="co">#&gt; Bifidobacterium.5_PT "Bifidobacterium.5_PT"</span>
<span class="co">#&gt; Bifidobacterium.6_PT "Bifidobacterium.6_PT"</span></code></pre></div>
<p>This is will be our <code>tax_table</code></p>
<p>Build a phyloseq object of theoretical composition</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps.th</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html">phyloseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">mck.otu</span>, taxa_are_rows <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>, 
                  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">mck.sam</span><span class="op">)</span>,
                  <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">mck.tax</span><span class="op">)</span><span class="op">)</span>

<span class="va">ps.th</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 55 taxa and 2 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 2 samples by 1 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 55 taxa by 6 taxonomic ranks ]</span></code></pre></div>
<p>The MIB mock contain 55 phylotypes. There are two types of mocks viz., MC3 and MC4</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_names-methods.html">sample_names</a></span><span class="op">(</span><span class="va">ps.th</span><span class="op">)</span>
<span class="co">#&gt; [1] "MC3" "MC4"</span></code></pre></div>
</div>
</div>
<div id="check-composition-of-theoretical" class="section level2">
<h2 class="hasAnchor">
<a href="#check-composition-of-theoretical" class="anchor"></a>Check composition of theoretical</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps.th.genus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/aggregate_taxa.html">aggregate_taxa</a></span><span class="op">(</span><span class="va">ps.th</span>, <span class="st">"Genus"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/plot_composition.html">plot_composition</a></span><span class="op">(</span><span class="va">ps.th.genus</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"right"</span>,
        legend.key.size<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.2</span>,<span class="st">'cm'</span><span class="op">)</span>,
        legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span><span class="op">(</span>col <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span><span class="op">(</span>ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="cExampleCustomMocks_files/figure-html/th-gen-1.png" width="3000"></p>
<p>These are not very useful to visualize with barplots. Too many genera!</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/plot_composition.html">plot_composition</a></span><span class="op">(</span><span class="va">ps.th.genus</span>, plot.type <span class="op">=</span> <span class="st">"heatmap"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_viridis.html">scale_fill_viridis_c</a></span><span class="op">(</span><span class="st">"Abudance (%)"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="cExampleCustomMocks_files/figure-html/th-heatmap-1.png" width="3000"></p>
</div>
<div id="new-experiment" class="section level2">
<h2 class="hasAnchor">
<a href="#new-experiment" class="anchor"></a>New experiment</h2>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Here example is store in the package.</span>
<span class="va">ps.mib.w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ps.mib.rds"</span>,
                        package<span class="op">=</span><span class="st">"chkMocks"</span>, mustWork <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#path for file</span>
<span class="va">ps.mib.w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="va">ps.mib.w</span><span class="op">)</span>

<span class="co"># taxa names are ASV seqs. Check first 2 names/ASV seqs</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps.mib.w</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>
<span class="co">#&gt; [1] "GTGTAACGCCTCCGAAGAGTCGCATGCTTTCACATGTTGTTCATTACATGTCAAGCCCAGGTAAGGTTCTTCGCGTTGCATCGAATTAAGCCACATACTCCACCGCTTGTGCGGGTCCCCNNNNNNNNNNCTTTGAGTTTTAATCTTGCGACCGTACTCCCCAGGCGGCACGCTTAACGCGTTAGCTCCGGCACGCAGGGGGTCGATTCCCCGCACACCAAGCGTGCACCGTTTACTGCCAGGACTACAG"</span>
<span class="co">#&gt; [2] "CTGTAGTCCTGGCAGTAAACGGTGCACGCTTGGTGTGCGGGGAATCGACCCCCTGCGTGCCGGAGCTAACGCGTTAAGCGTGCCGCCTGGGGAGTACGGTCGCAAGATTAAAACTCAAAGNNNNNNNNNNGGGGACCCGCACAAGCGGTGGAGTATGTGGCTTAATTCGATGCAACGCGAAGAACCTTACCTGGGCTTGACATGTAATGAACAACATGTGAAAGCATGCGACTCTTCGGAGGCGTTACAC"</span></code></pre></div>
</div>
<div id="assign-custom-taxonomy" class="section level2">
<h2 class="hasAnchor">
<a href="#assign-custom-taxonomy" class="anchor"></a>Assign custom taxonomy</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ps.mib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assignTaxonomyCustomMock.html">assignTaxonomyCustomMock</a></span><span class="op">(</span><span class="va">ps.mib.w</span>, <span class="co"># experimental mock community phyloseq</span>
                                   mock_db <span class="op">=</span> <span class="va">MIBTrainingSet</span>, <span class="co"># custome training set  </span>
                                   processors <span class="op">=</span> <span class="cn">NULL</span>,
                                   threshold <span class="op">=</span> <span class="fl">60</span>,
                                   strand <span class="op">=</span> <span class="st">"top"</span>,
                                   verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; Warning: Expected 8 pieces. Missing pieces filled with `NA` in 1351 rows [1, 3,</span>
<span class="co">#&gt; 5, 7, 8, 12, 14, 15, 16, 18, 19, 20, 21, 22, 24, 26, 28, 29, 30, 31, ...].</span></code></pre></div>
</div>
<div id="aggregate-to-species" class="section level2">
<h2 class="hasAnchor">
<a href="#aggregate-to-species" class="anchor"></a>Aggregate to species</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ps.mib</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/aggregate_taxa.html">aggregate_taxa</a></span><span class="op">(</span><span class="va">ps.mib</span>, <span class="st">"species"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/taxa_names-methods.html">taxa_names</a></span><span class="op">(</span><span class="va">ps.mib</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Micrococcus.2_PT"                    "Rhodococcus.1_PT"                   </span>
<span class="co">#&gt;  [3] "Bifidobacterium.10_PT"               "Bifidobacterium.4_PT"               </span>
<span class="co">#&gt;  [5] "Bifidobacterium.5_PT"                "Bifidobacterium.6_PT"               </span>
<span class="co">#&gt;  [7] "Bifidobacterium.8_PT"                "unclassified_Bifidobacterium"       </span>
<span class="co">#&gt;  [9] "Bacteroides.12_PT"                   "Bacteroides.13_PT"                  </span>
<span class="co">#&gt; [11] "Bacteroides.14_PT"                   "Bacteroides.15_PT"                  </span>
<span class="co">#&gt; [13] "unclassified_Bacteroides"            "Parabacteroides.17_PT"              </span>
<span class="co">#&gt; [15] "Prevotella.18_PT"                    "Alistipes.19_PT"                    </span>
<span class="co">#&gt; [17] "Bacillus.20_PT"                      "Bacillus.21_PT"                     </span>
<span class="co">#&gt; [19] "unclassified_Bacillus"               "Granulicatella.22_PT"               </span>
<span class="co">#&gt; [21] "Enterococcus.23_PT"                  "Lactobacillus.24_PT"                </span>
<span class="co">#&gt; [23] "Lactobacillus.25_PT"                 "Lactobacillus.26_PT"                </span>
<span class="co">#&gt; [25] "unclassified_Lactobacillus"          "Lactococcus.27_PT"                  </span>
<span class="co">#&gt; [27] "Streptococcus.28_PT"                 "Streptococcus.29_PT"                </span>
<span class="co">#&gt; [29] "unclassified_Streptococcus"          "Clostridium sensu stricto.30_PT"    </span>
<span class="co">#&gt; [31] "Anaerostipes.31_PT"                  "Blautia.32_PT"                      </span>
<span class="co">#&gt; [33] "Dorea.33_PT"                         "Lachnospiracea_incertae_sedis.34_PT"</span>
<span class="co">#&gt; [35] "Lachnospiracea_incertae_sedis.35_PT" "Lachnospiracea_incertae_sedis.36_PT"</span>
<span class="co">#&gt; [37] "Roseburia.37_PT"                     "Ruminococcus2.38_PT"                </span>
<span class="co">#&gt; [39] "Clostridium XI.39_PT"                "Clostridium IV.40_PT"               </span>
<span class="co">#&gt; [41] "Faecalibacterium.41_PT"              "Veillonella.42_PT"                  </span>
<span class="co">#&gt; [43] "Fusobacterium.43_PT"                 "Victivallis.44_PT"                  </span>
<span class="co">#&gt; [45] "Enterobacter.46_PT"                  "Enterobacter.48_PT"                 </span>
<span class="co">#&gt; [47] "Escherichia/Shigella.49_PT"          "Klebsiella.50_PT"                   </span>
<span class="co">#&gt; [49] "Salmonella.51_PT"                    "Serratia.52_PT"                     </span>
<span class="co">#&gt; [51] "Pseudomonas.53_PT"                   "Pseudomonas.54_PT"                  </span>
<span class="co">#&gt; [53] "unclassified_Pseudomonas"            "Akkermansia.55_PT"                  </span>
<span class="co">#&gt; [55] "Unknown"</span>

<span class="co"># convert to relative abundance  </span>
<span class="va">ps.mib</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/transform.html">transform</a></span><span class="op">(</span><span class="va">ps.mib</span>, <span class="st">"compositional"</span><span class="op">)</span></code></pre></div>
</div>
<div id="merge-with-theoretical" class="section level2">
<h2 class="hasAnchor">
<a href="#merge-with-theoretical" class="anchor"></a>Merge with theoretical</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># There is one sampe here</span>
<span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps.th</span><span class="op">)</span><span class="op">$</span><span class="va">MockType</span> <span class="op">&lt;-</span> <span class="st">"Theoretical"</span>

<span class="co"># adding new column to ps.mck.nw.tax may be for other comparisons user might be interested in doing.</span>
<span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps.mib</span><span class="op">)</span><span class="op">$</span><span class="va">MockType</span> <span class="op">&lt;-</span> <span class="st">"Experimental"</span>


<span class="va">ps.custom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/merge_phyloseq.html">merge_phyloseq</a></span><span class="op">(</span><span class="va">ps.mib</span>, <span class="va">ps.th</span><span class="op">)</span></code></pre></div>
<p>Compare the experimental mocks with theoretical mocks 3</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/compare2theorectical.html">compare2theorectical</a></span><span class="op">(</span><span class="va">ps.custom</span>,  theoretical_id <span class="op">=</span> <span class="st">"MC3"</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 7 x 2</span>
<span class="co">#&gt;   term            MC3</span>
<span class="co">#&gt;   &lt;chr&gt;         &lt;dbl&gt;</span>
<span class="co">#&gt; 1 Lib1_Mock_3  0.398 </span>
<span class="co">#&gt; 2 Lib1_Mock_4 -0.0880</span>
<span class="co">#&gt; 3 Lib2_Mock_3  0.405 </span>
<span class="co">#&gt; 4 Lib2_Mock_4 -0.0513</span>
<span class="co">#&gt; 5 Lib3_Mock_3  0.404 </span>
<span class="co">#&gt; 6 Lib3_Mock_4 -0.0711</span>
<span class="co">#&gt; 7 MC4          0.258</span></code></pre></div>
<p>Visualize</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cor.table.ref</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/compare2theorectical.html">compare2theorectical</a></span><span class="op">(</span><span class="va">ps.custom</span>, theoretical <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">corrr</span><span class="fu">::</span><span class="fu"><a href="https://corrr.tidymodels.org/reference/focus.html">focus</a></span><span class="op">(</span><span class="va">MC3</span><span class="op">)</span>
<span class="co">#&gt; Check that theoretical =  exists in sample_names(x)</span>
<span class="co">#&gt; returning all pairwise comparisons</span>

<span class="va">cor.table.ref</span> <span class="op">%&gt;%</span> 
  <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Remove MC4 theoretical and experimental and keep only those with MC3</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">term</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Lib1_Mock_4"</span>, <span class="st">"Lib2_Mock_4"</span>, <span class="st">"Lib3_Mock_4"</span>, <span class="st">"MC4"</span><span class="op">)</span> <span class="op">&amp;</span>
                  <span class="va">variable</span> <span class="op">==</span> <span class="st">"MC3"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">value</span>,<span class="va">term</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span>fill<span class="op">=</span><span class="st">"steelblue"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">#facet_grid(~variable) +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Experimental Mocks"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Spearman correlation"</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">#scale_fill_viridis_c() +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Species level correlation"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_continuous</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Using term as id variables</span></code></pre></div>
<p><img src="cExampleCustomMocks_files/figure-html/viz-cor-1-1.png" width="3000"></p>
<p>This is an example where one can observe that in high diversity mocks with some closely related “species”, the assignments based on short reads is difficult. The correlation values are less than 0.5.</p>
</div>
<div id="aggregate-to-genus" class="section level2">
<h2 class="hasAnchor">
<a href="#aggregate-to-genus" class="anchor"></a>Aggregate to Genus</h2>
<p>It is well known that the Genus level assignments with short reads is better than “species” level assignments. Therefore, we can check the correlation at genus level.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ps.custom.gen</span> <span class="op">&lt;-</span> <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/aggregate_taxa.html">aggregate_taxa</a></span><span class="op">(</span><span class="va">ps.custom</span>, <span class="st">"genus"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/compare2theorectical.html">compare2theorectical</a></span><span class="op">(</span><span class="va">ps.custom.gen</span>,  theoretical_id <span class="op">=</span> <span class="st">"MC3"</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 7 x 2</span>
<span class="co">#&gt;   term             MC3</span>
<span class="co">#&gt;   &lt;chr&gt;          &lt;dbl&gt;</span>
<span class="co">#&gt; 1 Lib1_Mock_3  0.771  </span>
<span class="co">#&gt; 2 Lib1_Mock_4 -0.00647</span>
<span class="co">#&gt; 3 Lib2_Mock_3  0.762  </span>
<span class="co">#&gt; 4 Lib2_Mock_4 -0.0521 </span>
<span class="co">#&gt; 5 Lib3_Mock_3  0.766  </span>
<span class="co">#&gt; 6 Lib3_Mock_4 -0.0447 </span>
<span class="co">#&gt; 7 MC4          0.0807</span></code></pre></div>
<p>Compare the values for MC3</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cor.table.ref</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/compare2theorectical.html">compare2theorectical</a></span><span class="op">(</span><span class="va">ps.custom.gen</span>, theoretical <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">corrr</span><span class="fu">::</span><span class="fu"><a href="https://corrr.tidymodels.org/reference/focus.html">focus</a></span><span class="op">(</span><span class="va">MC3</span><span class="op">)</span>
<span class="co">#&gt; Check that theoretical =  exists in sample_names(x)</span>
<span class="co">#&gt; returning all pairwise comparisons</span>

<span class="va">cor.table.ref</span> <span class="op">%&gt;%</span> 
  <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Remove MC4 theoretical and experimental and keep only those with MC3</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">term</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Lib1_Mock_4"</span>, <span class="st">"Lib2_Mock_4"</span>, <span class="st">"Lib3_Mock_4"</span>, <span class="st">"MC4"</span><span class="op">)</span> <span class="op">&amp;</span>
                  <span class="va">variable</span> <span class="op">==</span> <span class="st">"MC3"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">value</span>,<span class="va">term</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span>fill<span class="op">=</span><span class="st">"steelblue"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">#facet_grid(~variable) +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Experimental Mocks"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Spearman correlation"</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">#scale_fill_viridis_c() +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Genus level correlation"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_continuous</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Using term as id variables</span></code></pre></div>
<p><img src="cExampleCustomMocks_files/figure-html/viz-cor-2-1.png" width="3000"></p>
<p>Compare the values for MC4</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cor.table.ref</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/compare2theorectical.html">compare2theorectical</a></span><span class="op">(</span><span class="va">ps.custom.gen</span>, theoretical <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu">corrr</span><span class="fu">::</span><span class="fu"><a href="https://corrr.tidymodels.org/reference/focus.html">focus</a></span><span class="op">(</span><span class="va">MC4</span><span class="op">)</span>
<span class="co">#&gt; Check that theoretical =  exists in sample_names(x)</span>
<span class="co">#&gt; returning all pairwise comparisons</span>

<span class="va">cor.table.ref</span> <span class="op">%&gt;%</span> 
  <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="co"># Remove MC4 theoretical and experimental and keep only those with MC3</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="va">term</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Lib1_Mock_3"</span>, <span class="st">"Lib2_Mock_3"</span>, <span class="st">"Lib3_Mock_3"</span>, <span class="st">"MC3"</span><span class="op">)</span> <span class="op">&amp;</span>
                  <span class="va">variable</span> <span class="op">==</span> <span class="st">"MC4"</span><span class="op">)</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">value</span>,<span class="va">term</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span>fill<span class="op">=</span><span class="st">"steelblue"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">#facet_grid(~variable) +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Experimental Mocks"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Spearman correlation"</span><span class="op">)</span> <span class="op">+</span>
  <span class="co">#scale_fill_viridis_c() +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Genus level correlation"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_continuous</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Using term as id variables</span></code></pre></div>
<p><img src="cExampleCustomMocks_files/figure-html/viz-cor-3-1.png" width="3000"></p>
<p>There is a major improvement in correlation between theoretical and expected mock communities.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sudarshan Shetty.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
