<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction • chkMocks</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="chkMocks">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">chkMocks</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microsud/chkMocks/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Introduction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/microsud/chkMocks/blob/master/vignettes/Introduction.Rmd"><code>vignettes/Introduction.Rmd</code></a></small>
      <div class="hidden name"><code>Introduction.Rmd</code></div>

    </div>

    
    
<p>Every microbiome sequencing experiment must have a positive control. However, how do we make use of these mock controls to guide our quality check is not easily available. A basic question to asks, <em>Is the composition in experimental mock standards similar to theoretical expected composition?</em><br>
We can visually compare the composition bar-plots and check for correlation between experimental and theoretical community composition.<br><code>chkMocks</code> eases these basic comparisons.<br>
Note: Currently, only ZymoBiomics mocks are supported and we are working on adding support for custom databases.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microsud/chkMocks">chkMocks</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com">patchwork</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'patchwork' was built under R version 4.0.5</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span></code></pre></div>
<div id="zymobiomics" class="section level2">
<h2 class="hasAnchor">
<a href="#zymobiomics" class="anchor"></a>ZymoBiomics</h2>
<p>Before starting the analysis you need:</p>
<ul>
<li>A phyloseq object with mock community composition and must have <code>taxa_names</code> as ASV seqs</li>
</ul>
<p>Here, the example data are from Karstens L, Asquith M, Davin S, Fair D, Gregory WT, Wolfe AJ, Braun J, McWeeney S. 2019. <a href="https://doi.org/10.1128/mSystems.00290-19">Controlling for contaminants in low-biomass 16S rRNA gene sequencing experiments</a>. mSystems 4:e00290-19.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ZymoExamplePseq</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 946 taxa and 9 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 9 samples by 6 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 946 taxa by 6 taxonomic ranks ]</span></code></pre></div>
<p>A single function, <code>checkZymoBiomics</code> will do the following:</p>
<p>Take an input phyloseq object of mock communities with <code>taxa_names</code> as ASV seqs and <code>ZymoTrainingSet</code> to assign taxonomy. The <code>ZymoTrainingSet</code> is a with only the full-lenght 16S rRNA gene sequences of the candidates in ZymoBIOMICS™ Microbial Community Standard. <a href="https://files.zymoresearch.com/protocols/_d6300_zymobiomics_microbial_community_standard.pdf">Catalog No. D6300</a>. ASVs unrelated to <code>ZymoTrainingSet</code> are unclassified. The function will return two phyloseq objects a) ASVs level and b) Agglomerated at Species level. The agglomerated species level data is used to check for correlation with theoretical composition.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">output.dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkZymoBiomics.html">checkZymoBiomics</a></span><span class="op">(</span><span class="va">ZymoExamplePseq</span>,
                               mock_db <span class="op">=</span> <span class="va">ZymoTrainingSet</span>,
                               multithread<span class="op">=</span> <span class="fl">2</span>,
                               threshold <span class="op">=</span> <span class="fl">80</span>,
                               verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; Using internal reference database</span>
<span class="co">#&gt;             for ZymoBiomics</span>
<span class="co">#&gt; Warning: Expected 8 pieces. Missing pieces filled with `NA` in 936 rows [9, 11,</span>
<span class="co">#&gt; 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, ...].</span></code></pre></div>
<p>Get the agglomerated species level data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ps_species</span> <span class="op">&lt;-</span> <span class="va">output.dat</span><span class="op">$</span><span class="va">ps_species</span>

<span class="va">ps_species</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 9 taxa and 10 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 10 samples by 8 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 9 taxa by 15 taxonomic ranks ]</span></code></pre></div>
<p>Check assignments</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/get_taxa_unique.html">get_taxa_unique</a></span><span class="op">(</span><span class="va">output.dat</span><span class="op">$</span><span class="va">ps_species</span>, <span class="st">"Species"</span><span class="op">)</span>
<span class="co">#&gt; [1] "Bacillus.subtilis"       "Listeria.monocytogenes" </span>
<span class="co">#&gt; [3] "Staphylococcus.aureus"   "Enterococcus.faecalis"  </span>
<span class="co">#&gt; [5] "Lactobacillus.fermentum" "Escherichia.coli"       </span>
<span class="co">#&gt; [7] "Salmonella.enterica"     "Pseudomonas.aeruginosa" </span>
<span class="co">#&gt; [9] NA</span></code></pre></div>
<p>Plot composition</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/plotZymoDefault.html">plotZymoDefault</a></span><span class="op">(</span><span class="va">output.dat</span><span class="op">)</span> 
<span class="co">#&gt; Joining, by = "FeatureID"</span>
<span class="co">#&gt; Warning in class(x) &lt;- c(setdiff(subclass, tibble_class), tibble_class): Setting</span>
<span class="co">#&gt; class(x) to multiple strings ("tbl_df", "tbl", ...); result will no longer be an</span>
<span class="co">#&gt; S4 object</span>
<span class="co">#&gt; Joining, by = "chk.names"</span></code></pre></div>
<p><img src="Introduction_files/figure-html/unnamed-chunk-6-1.png" width="2400"></p>
<p>The above plot demonstrates how well/bad the experimental mocks behaved compared to theoretical composition.<br>
The data from Karstens L, et al., 2019. <a href="https://doi.org/10.1128/mSystems.00290-19">mSystems</a> started from cells-&gt;DNA extraction-&gt;pcr amplification-&gt;sequencing. At every step there is a possibility for bias. This is demonstrated with undiluted mock sample showing low correlation. Additionally, diluted samples have several unknown, potential contaminants common to low-biomass samples.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/sessioninfo/man/session_info.html">session_info</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; - Session info ---------------------------------------------------------------</span>
<span class="co">#&gt;  setting  value                       </span>
<span class="co">#&gt;  version  R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt;  os       Windows 10 x64              </span>
<span class="co">#&gt;  system   x86_64, mingw32             </span>
<span class="co">#&gt;  ui       RTerm                       </span>
<span class="co">#&gt;  language (EN)                        </span>
<span class="co">#&gt;  collate  English_Netherlands.1252    </span>
<span class="co">#&gt;  ctype    English_Netherlands.1252    </span>
<span class="co">#&gt;  tz       Europe/Berlin               </span>
<span class="co">#&gt;  date     2021-04-29                  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; - Packages -------------------------------------------------------------------</span>
<span class="co">#&gt;  ! package              * version  date       lib source        </span>
<span class="co">#&gt;    ade4                   1.7-16   2020-10-28 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    ape                    5.4-1    2020-08-13 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    Biobase                2.50.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    BiocGenerics           0.36.1   2021-04-16 [2] Bioconductor  </span>
<span class="co">#&gt;    BiocParallel           1.24.1   2020-11-06 [2] Bioconductor  </span>
<span class="co">#&gt;    biomformat             1.18.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    Biostrings             2.58.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    bit                    4.0.4    2020-08-04 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    bit64                  4.0.5    2020-08-30 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    bitops                 1.0-6    2013-08-17 [2] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    blob                   1.2.1    2020-01-20 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    bslib                  0.2.4    2021-01-25 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    cachem                 1.0.4    2021-02-13 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    callr                  3.7.0    2021-04-20 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    chkMocks             * 0.1.0    2021-04-29 [2] local         </span>
<span class="co">#&gt;    cli                    2.4.0    2021-04-05 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    cluster                2.1.0    2019-06-19 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    codetools              0.2-16   2018-12-24 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    colorspace             1.4-1    2019-03-18 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    corrr                  0.4.3    2020-11-24 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    crayon                 1.4.1    2021-02-08 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    dada2                  1.18.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    data.table             1.13.0   2020-07-24 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    DBI                    1.1.1    2021-01-15 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    DECIPHER               2.18.1   2020-10-29 [2] Bioconductor  </span>
<span class="co">#&gt;    DelayedArray           0.16.3   2021-03-24 [2] Bioconductor  </span>
<span class="co">#&gt;    desc                   1.3.0    2021-03-05 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    devtools               2.4.0    2021-04-07 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    digest                 0.6.27   2020-10-24 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    dplyr                * 1.0.2    2020-08-18 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    ellipsis               0.3.1    2020-05-15 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    evaluate               0.14     2019-05-28 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    fansi                  0.4.2    2021-01-15 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    farver                 2.1.0    2021-02-28 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    fastmap                1.0.1    2019-10-08 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    foreach                1.5.1    2020-10-15 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    fs                     1.5.0    2020-07-31 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    generics               0.1.0    2020-10-31 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    GenomeInfoDb           1.26.7   2021-04-09 [2] Bioconductor  </span>
<span class="co">#&gt;    GenomeInfoDbData       1.2.4    2020-10-19 [2] Bioconductor  </span>
<span class="co">#&gt;    GenomicAlignments      1.26.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    GenomicRanges          1.42.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    ggplot2              * 3.3.3    2020-12-30 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    glue                   1.4.2    2020-08-27 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    gtable                 0.3.0    2019-03-25 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    highr                  0.9      2021-04-16 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    hms                    1.0.0    2021-01-13 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    htmltools              0.5.1.1  2021-01-22 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    hwriter                1.3.2    2014-09-10 [2] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    igraph                 1.2.6    2020-10-06 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    IRanges                2.24.1   2020-12-12 [2] Bioconductor  </span>
<span class="co">#&gt;    iterators              1.0.13   2020-10-15 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    jpeg                   0.1-8.1  2019-10-24 [2] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    jquerylib              0.1.3    2020-12-17 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    jsonlite               1.7.1    2020-09-07 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    knitr                  1.32     2021-04-14 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    labeling               0.4.2    2020-10-20 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    lattice                0.20-41  2020-04-02 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    latticeExtra           0.6-29   2019-12-19 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    lifecycle              1.0.0    2021-02-15 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    magrittr               2.0.1    2020-11-17 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    MASS                   7.3-53   2020-09-09 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    Matrix                 1.2-18   2019-11-27 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    MatrixGenerics         1.2.1    2021-01-30 [2] Bioconductor  </span>
<span class="co">#&gt;    matrixStats            0.58.0   2021-01-29 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    memoise                2.0.0    2021-01-26 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    mgcv                   1.8-33   2020-08-27 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    microbiome             1.12.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    multtest               2.46.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    munsell                0.5.0    2018-06-12 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    nlme                   3.1-149  2020-08-23 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    patchwork            * 1.1.1    2020-12-17 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    permute                0.9-5    2019-03-12 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    phyloseq             * 1.34.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    pillar                 1.6.0    2021-04-13 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    pkgbuild               1.2.0    2020-12-15 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    pkgconfig              2.0.3    2019-09-22 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    pkgdown                1.6.1    2020-09-12 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    pkgload                1.2.1    2021-04-06 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    plyr                   1.8.6    2020-03-03 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    png                    0.1-7    2013-12-03 [2] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    prettyunits            1.1.1    2020-01-24 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    processx               3.5.1    2021-04-04 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    progress               1.2.2    2019-05-16 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    ps                     1.6.0    2021-02-28 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    purrr                  0.3.4    2020-04-17 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    R6                     2.5.0    2020-10-28 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    ragg                   1.1.2    2021-03-17 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    RColorBrewer           1.1-2    2014-12-07 [2] CRAN (R 4.0.0)</span>
<span class="co">#&gt;    Rcpp                   1.0.5    2020-07-06 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;  D RcppParallel           5.1.2    2021-04-15 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    RCurl                  1.98-1.3 2021-03-16 [2] CRAN (R 4.0.4)</span>
<span class="co">#&gt;    remotes                2.3.0    2021-04-01 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    reshape2               1.4.4    2020-04-09 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    rhdf5                  2.34.0   2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;  D rhdf5filters           1.2.0    2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    Rhdf5lib               1.12.1   2021-01-26 [2] Bioconductor  </span>
<span class="co">#&gt;    rlang                  0.4.10   2020-12-30 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    rmarkdown              2.7      2021-02-19 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    rprojroot              2.0.2    2020-11-15 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    Rsamtools              2.6.0    2020-10-27 [2] Bioconductor  </span>
<span class="co">#&gt;    RSQLite                2.2.1    2020-09-30 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    Rtsne                  0.15     2018-11-10 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    S4Vectors              0.28.1   2020-12-09 [2] Bioconductor  </span>
<span class="co">#&gt;    sass                   0.3.1    2021-01-24 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    scales                 1.1.1    2020-05-11 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    sessioninfo            1.1.1    2018-11-05 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    ShortRead              1.48.0   2020-10-28 [2] Bioconductor  </span>
<span class="co">#&gt;    stringi                1.5.3    2020-09-09 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    stringr                1.4.0    2019-02-10 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    SummarizedExperiment   1.20.0   2020-10-28 [2] Bioconductor  </span>
<span class="co">#&gt;    survival               3.2-7    2020-09-28 [3] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    systemfonts            1.0.1    2021-02-09 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    testthat               3.0.2    2021-02-14 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    textshaping            0.3.3    2021-03-16 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    tibble                 3.0.4    2020-10-12 [2] CRAN (R 4.0.3)</span>
<span class="co">#&gt;    tidyr                  1.1.2    2020-08-27 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    tidyselect             1.1.0    2020-05-11 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    usethis                2.0.1    2021-02-10 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    utf8                   1.2.1    2021-03-12 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    vctrs                  0.3.4    2020-08-29 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    vegan                  2.5-7    2020-11-28 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    viridisLite            0.4.0    2021-04-13 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    withr                  2.4.2    2021-04-18 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    xfun                   0.22     2021-03-11 [2] CRAN (R 4.0.5)</span>
<span class="co">#&gt;    XVector                0.30.0   2020-10-28 [2] Bioconductor  </span>
<span class="co">#&gt;    yaml                   2.2.1    2020-02-01 [2] CRAN (R 4.0.2)</span>
<span class="co">#&gt;    zlibbioc               1.36.0   2020-10-28 [2] Bioconductor  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [1] C:/Users/sudar/AppData/Local/Temp/RtmpwpIp5Q/temp_libpath18d8491b328a</span>
<span class="co">#&gt; [2] C:/Users/sudar/Documents/R/win-library/4.0</span>
<span class="co">#&gt; [3] C:/Program Files/R/R-4.0.3/library</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  D -- DLL MD5 mismatch, broken installation.</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sudarshan Shetty.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
